pkg/util/conditions/condition_test.go

```diff
@@ -0,0 +1,265 @@
+package conditions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	arofake "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/fake"
+	"github.com/Azure/ARO-RP/pkg/util/cmp"
+)
+
+func TestSetCondition(t *testing.T) {
```

Comment:
I think we need the following test cases:

* Case where we exit early and do not call `UpdateStatus` (is `noop` case supposed to test this? See [this thread](https://github.com/Azure/ARO-RP/pull/1508/files#r656952649))
* Case where we set a new condition (current conditions slice is empty, but expected result would be a new condition from the `input`).
* Case where an existing condition gets updated with the condition which is exactly the same: we want to make sure that transition time doesn't change